gen/ivory.hoon

```diff
@@ -1,56 +1,61 @@
 ::
-::::  /hoon/metal/gen
+::::  /hoon/ivory/gen
   ::
 /?    310
 ::
 ::::
   !:
 :-  %say
-|=  $:  {now/@da * bec/beak}
-        *
-    ==
+|=  [[now=@da * bec=beak] *]
 :-  %noun
-::
 ::  sys: root path to boot system, `/~me/[desk]/now/sys`
 ::
-=+  sys=`path`/(scot %p p.bec)/[q.bec]/(scot %da now)/sys
-::
+=/  sys=path
+  /(scot %p p.bec)/[q.bec]/(scot %da now)/sys
 ::  compiler-source: hoon source file producing compiler, `sys/hoon`
 ::
-=+  compiler-source=.^(@t %cx (welp sys /hoon/hoon))
-::
+=/  compiler-source
+  .^(@t %cx (welp sys /hoon/hoon))
 ::  compiler-hoon: compiler as hoon expression
 ::
+::    We parse with +ream for reproducibility.
+::
 ~&  %ivory-parsing
-=+  compiler-hoon=(ream compiler-source)
+=/  compiler-hoon  (ream compiler-source)
 ~&  %ivory-parsed
-::
 ::  arvo-source: hoon source file producing arvo kernel, `sys/arvo`
 ::
-=+  arvo-source=.^(@t %cx (welp sys /arvo/hoon))
-::
+=/  arvo-source
+  .^(@t %cx (welp sys /arvo/hoon))
 ::  whole-hoon: arvo within compiler
 ::
-=+  whole-hoon=`hoon`[%tsbn compiler-hoon [%tsbn [%$ 7] (ream arvo-source)]]
-::
+=/  whole-hoon=hoon
+  :+  %tsbn  compiler-hoon
+  :+  %tsld  (ream arvo-source)
+  [%$ 7]
 ::  compile the whole schmeer
 ::
 ~&  %ivory-compiling
-=+  whole-formula=q:(~(mint ut %noun) %noun whole-hoon)
+=/  whole-formula
+  q:(~(mint ut %noun) %noun whole-hoon)
 ~&  %ivory-compiled
-::
 ::  zuse-ovo: standard library installation event
 ::
-=/  zuse-ovo=ovum
-  =/  pax  (weld sys /zuse)
-  =/  txt  .^(@ %cx (weld pax /hoon))
-  [/vane/zuse [%veer %$ pax txt]]
+::    Arvo parses the %veer payload with +rain, so
```

Comment:
It would be better to pass the spur than the empty path. Otherwise if an error occurs, we won't know which file errored.

Reply:
The spur, flopped? `/sys/zuse/hoon`?